### PR TITLE
52204 - create products txt file

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
Code changes and Unit tests
Functional Test

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 1]
```diff
@@ -15,7 +15,8 @@ public static class FileContentHelper
         private const string FileContent_avcs = "AVCS";
         private const string FileContent_base = "Base";
         private const string FileContent_dvd = "Media','DVD_SERVICE'";
-
+        public static string dirName, baseFolderNumber;
+        ////static string baseFolderNumber;
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs:19]: remove this commented code

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs [hunk 2]
```diff
@@ -126,7 +126,7 @@ private SalesCatalogueProductResponse GetSalesCatalogueResponse()
                         FileSize = 400,
                         Bundle = new List<Bundle>
                         {
-                            new Bundle{BundleType = 0,Location = "M1:B1"}
+                            new Bundle{BundleType = "0",Location = "M1:B1"}
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentDataServiceTest.cs]: Resolved

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs (hunk 1 above)
Comment (MastekRohitChettri) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs]: Resolved

**Review by harshal11869 — APPROVED** [review]
LGTM

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static class FileContentHelper
         private const string FileContent_avcs = "AVCS";
         private const string FileContent_base = "Base";
         private const string FileContent_dvd = "Media','DVD_SERVICE'";
-
+        public static string dirName, baseFolderNumber;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs:18]: Are these used? If not remove them please.

**Review by ajasonca — CHANGES_REQUESTED** [review]
Mostly good, a couple of var scoping issues and some redundant var declarations to remove and it should be good to go.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 4]
```diff
@@ -16,7 +17,12 @@ public class ExchangeSetForLargeFile
         private TestConfiguration Config { get; set; }
         public DataHelper DataHelper { get; set; }
         private HttpResponseMessage ApiEssResponse { get; set; }
-        private string DownloadedFolderPath { get; set; }
+
+        private List<string> DownloadedFolderPath;
+
```
[the remark this answers is not in the document]
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:22]: Updated. Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs (hunk 3 above)
Comment (Arit14975) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs]: Updated. Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 5]
```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
...
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderIsGenerated()
         {
-            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            int mediaNumber = 1;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:72]: I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 6]
```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
...
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAReadMeTxtFileIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAReadMeTxtFileIsGenerated()
         {
+            int mediaNumber = 1;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:86]: I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 7]
```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
...
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenASerialEncFileIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenASerialEncFileIsGenerated()
         {
+            int mediaNumber = 1;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:113]: I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 8]
```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
...
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenEncFilesAreGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenEncFilesAreGenerated()
         {
+            int mediaNumber = 1;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:140]: I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 9]
```diff
@@ -157,12 +164,29 @@ public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenEncF
...
+
+                mediaNumber++;
+            }
+        }
+
+        [Test]
+        [Category("SmokeTest")]
+        public void WhenICallExchangeSetApiWithAnInValidProductVersion_ThenAProductTxtFileIsGenerated()
+        {
+            int mediaNumber = 1;
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs:178]: I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 10]
```diff
@@ -366,5 +345,42 @@ public static void CheckSerialEncFileContentForLargeMediaExchangeSet(string inpu
...
+                Assert.AreEqual(200, (int)apiEssResponse.StatusCode, $"Incorrect status code is returned {apiEssResponse.StatusCode}, instead of the expected status 200.");
+
+                var finalBatchStatusUrl = $"{Config.FssConfig.BaseUrl}/batch/62713adc-6999-40f6-86b1-ca08ab693d44/status"; //here BatchId is hardcoded and will be made dynamic in future
+
+                var batchStatus = await FssBatchHelper.CheckBatchIsCommitted(finalBatchStatusUrl, FssJwtToken);
+                Assert.AreEqual("Committed", batchStatus, $"Incorrect batch status is returned {batchStatus} for url {finalBatchStatusUrl}, instead of the expected status Committed.");
+
+                var downloadFileUrl = $"{Config.FssConfig.BaseUrl}/batch/62713adc-6999-40f6-86b1-ca08ab693d44/files/{FolderName}.zip"; //here BatchId is hardcoded and will be made dynamic in future
+
+                var extractDownloadedFolder = await FssBatchHelper.ExtractDownloadedFolderForLargeFiles(downloadFileUrl.ToString(), FssJwtToken, FolderName);
```
Comment (ajasonca) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs:364]: downloadFileUrl is a string, so the .ToString() is redundant and can be removed.

**Review by ajasonca — CHANGES_REQUESTED** [review]
Sorry, a couple more things I've noticed since the last changes

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs [hunk 11]
```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
...
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderIsGenerated()
         {
-            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            int mediaNumber = 1;
```
[the remark this answers is not in the document]
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs (hunk 5 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs (hunk 6 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs (hunk 7 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs (hunk 8 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs (hunk 9 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs (hunk 10 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs]: Updated.

Thanks

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs [hunk 12]
```diff
@@ -15,7 +15,6 @@ public static class FileContentHelper
         private const string FileContent_avcs = "AVCS";
         private const string FileContent_base = "Base";
         private const string FileContent_dvd = "Media','DVD_SERVICE'";
-
         private static TestConfiguration Config = new TestConfiguration();
         private static FssApiClient FssApiClient = new FssApiClient();
         public static async Task<string> CreateExchangeSetFile(HttpResponseMessage apiEssResponse, string FssJwtToken)
```
Comment (harshal11869) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs:20]: downloadFileUrl .ToString() can be removed line no. 35

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs (hunk 12 above)
Comment (Mayuresh-Gawde) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/FileContentHelper.cs:20]: As discussed, this is the existing ESS FT code used for small\medium update sets. Hence, we haven't changed that.

Thanks..

**Review by harshal11869 — APPROVED** [review]

**Review by ajasonca — APPROVED** [review]
All looking good

**Pull request merged**